src/lib/mina_base/user_command.ml

```diff
@@ -165,42 +166,42 @@ module Make_to_all_verifiable
 struct
   let to_all_verifiable (ts : t Strategy.Command_wrapper.t list) ~load_vk_cache
       : Verifiable.t Strategy.Command_wrapper.t list Or_error.t =
+    (* TODO: it seems we're just doing noop with the Strategy.Command_wrapper,
+       need double-check.
+    *)
     let open Or_error.Let_syntax in
-    (* First we tag everything with its index *)
-    let its = List.mapi ts ~f:(fun i x -> (i, x)) in
-    (* then we partition out the zkapp commands *)
-    let izk_cmds, is_cmds =
-      List.partition_map its ~f:(fun (i, cmd) ->
-          match Strategy.Command_wrapper.unwrap cmd with
-          | Zkapp_command c ->
-              First (i, Strategy.Command_wrapper.map cmd ~f:(Fn.const c))
-          | Signed_command c ->
-              Second (i, Strategy.Command_wrapper.map cmd ~f:(Fn.const c)) )
+    let partitioner (cmd : t Strategy.Command_wrapper.t) =
+      match Strategy.Command_wrapper.unwrap cmd with
+      | Zkapp_command c ->
+          First (Strategy.Command_wrapper.map cmd ~f:(Fn.const c))
+      | Signed_command c ->
+          Second (Strategy.Command_wrapper.map cmd ~f:(Fn.const c))
     in
-    (* then unzip the indices *)
-    let ixs, zk_cmds = List.unzip izk_cmds in
-    (* then we verify the zkapp commands *)
-    (* TODO: we could optimize this by skipping the fee payer and non-proof authorizations *)
-    let accounts_referenced =
-      List.fold_left zk_cmds ~init:Account_id.Set.empty ~f:(fun set zk_cmd ->
-          Strategy.Command_wrapper.unwrap zk_cmd
-          |> Zkapp_command.accounts_referenced |> Account_id.Set.of_list
-          |> Set.union set )
+    let process_left zk_cmds =
```

Comment:
Do we have any existing unit tests that can confirm new logic is correct?

Reply:
Let me check

Reply:
Unfortunately no, do you want tests? We could do it in another PR I guess.